utils/visualize.py

```diff
@@ -0,0 +1,141 @@
+from distutils.util import convert_path
+import networkx as nx
+import matplotlib.pyplot as plt
+from torch_geometric.data import Data, HeteroData
+from torch_geometric.utils import to_networkx
+from torch import Tensor
+import torch
+
+from typing import Union, Optional
+
+
+val_map = {"customer": 1.0, "article": 0.35}
+
+
+def select_layout(G: Union[nx.DiGraph, nx.Graph], node_types: Optional[list]):
+    one_category_nodes = [
+        node if node_type == "customer" else None
+        for node, node_type in zip(G.nodes(), node_types)
+    ]
+
+    pos = nx.bipartite_layout(G, nodes=one_category_nodes)
+    # pos = nx.kamada_kawai_layout(G)
```

Comment:
can we remove these?:)

Reply:
These are on purposely left in, it's useful to see various versions of the graph. We could of course have an if statement and pass in a selector variable but I think for our current purposes this is faster and more managable.